lib/active_hash/base.rb

```diff
@@ -127,9 +127,17 @@ def data=(array_of_hashes)
         end
       end
 
-      def exists?(record)
-        if record.id.present?
-          record_index[record.id.to_s].present?
+      def exists?(args = nil)
+        if args.respond_to?(:id)
+          record_index[args.id.to_s].present?
+        elsif args === false
+          false
+        elsif args.nil?
+          all.count.positive?
+        elsif args.class == Hash
+          all.where(args).count.positive?
```

Comment:
is there a reason to pick `all.where(args).count.positive?` vs `all.where(args).present?`

Reply:
The reason why I picked it is I misunderstood `where(args)` like AR object. 
I tested both by the follow script, then I know `present?` is faster than `count.positive?` 
So I haver replaced 'count.positive?` with `present?`.

```ruby
require 'active_hash'

class Country < ActiveHash::Base
end

20000.times do
  Country.create
end

puts 'count.positive?'
before = Time.now
Country.count.positive?
after = Time.now
puts after - before

puts 'present?'
before = Time.now
Country.present?
after = Time.now
puts after - before

$ bundle exec ruby test.rb
count.positive?
7.0e-06
present?
3.0e-06
```